README.md

```diff
@@ -362,12 +362,16 @@ This class extends [`EventEmitter`][] from Node.js.
     only makes sense to specify if there is some kind of asynchronous component
     to the task. Keep in mind that Worker threads are generally not built for
     handling I/O in parallel.
-  * `useAtomics`: (`boolean`) Use the [`Atomics`][] API for faster communication
+  * `atomics`: (`sync` | `async` | `disabled`) Use the [`Atomics`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Atomics) API for faster communication
     between threads. This is on by default. You can disable `Atomics` globally by
-    setting the environment variable `PISCINA_DISABLE_ATOMICS` to `1`. 
-    If `useAtomics` is `true`, it will cause to pause threads (stoping all execution)
-    between tasks. Ideally, threads should wait for all operations to finish before 
-    returning control to the main thread (avoid having open handles within a thread).
+    setting the environment variable `PISCINA_DISABLE_ATOMICS` to `1` .
+    If `atomics` is `sync`, it will cause to pause threads (stoping all execution)
+    between tasks. Ideally, threads should wait for all operations to finish before
+    returning control to the main thread (avoid having open handles within a thread). If still want to have the possibility
+    of having open handles or handle asynchrnous tasks, you can set the environment variable `PISCINA_ENABLE_ASYNC_ATOMICS` to `1` or setting `options.atomics` to `async`.
+
+   > **Note**: The `async` mode comes with performance penalties and can lead to undesired behaviour if open handles are not tracked correctly.
```

Comment:
Can you explain the open handle thing? Why is that an issue with async?

Reply:
Mostly as we are not doing event loop ticks, these can add some overhead if the workers start doing more than it should.

Even waiting for just the worker to wake up added some ticks that showed a difference of ±5% which is really negligible.

I've not strong opinion in this one; but if we want to make this the default behaviour we should document that it is important to not keep the workers doing too much background tasks as they can face some penalty (as well if using `async`, implementer needs to account for the teardown of the worker when idle, i.e. no more tasks scheduled)

Reply:
If that sounds good, I can adjust the PR for that 👍